components/autofill/src/db/store.rs

```diff
@@ -48,10 +49,12 @@ pub struct Store {
 }
 
 impl Store {
-    pub fn new(db_path: impl AsRef<Path>) -> Result<Self> {
-        Ok(Self {
-            db: Mutex::new(AutofillDb::new(db_path)?),
-        })
+    pub fn new(db_path: impl AsRef<Path>) -> ApiResult<Self> {
+        handle_error! {
```

Comment:
I like this pattern, and the more I see it I wonder if we could have a proc macro attribute like:

```rs
#[handle_err(AutofillApiError)]
pub fn new(db_path: impl AsRef<Path>) -> Result<Self> {
    Ok(Self {
        db: Mutex::new(AutofillDb::new(db_path)?),
    })
}
```

That has the same semantics as the `handle_error!` - it's orthogonal to this PR but thought it would be interesting to think out loud

Reply:
Yeah, I think that could a nice improvement.

One question is how should the return type read should this be `-> Result<>` or `-> ApiResult<>`?  The code itself is creating a `Result<>`, but then the proc-macro is turning it into an `ApiResult<>`.

However we answered that, I think it would be better than the current macro which I think assumes there's a `Result<>` type imported in the macro and will error out in a hard to understand way if there's not.

Reply:
I'm indifferent - If we had it as a `Result` then _maybe_ we don't need an `ApiResult`? maybe the macro can replace whatever the function returns with a `Result<T, AutofillApiError>` since proc macro attributes can modify function tokens 🤷 

Regardless it'll might also nice because we can also add some error handling to print nicer messages too

Reply:
Created https://github.com/mozilla/application-services/issues/5183 in case we want to do this!

Reply:
#5184 😬